File: src/main/java/net/matees/settings/global/FlightEnabled.java
package net.matees.settings.global;

import org.bukkit.Bukkit;
import org.bukkit.Material;

import net.matees.settings.BooleanSetting;
import net.matees.settings.Global;

public class FlightEnabled extends BooleanSetting implements Global {

    private static final FlightEnabled INSTANCE = new FlightEnabled();
    private Boolean setting = false;

    private FlightEnabled() {
    }

    public static FlightEnabled getInstance() {
        return INSTANCE;
    }

    @Override
    public void onChange() {
        Bukkit.getOnlinePlayers().forEach(player -> player.setAllowFlight(this.getSetting()));
    }

    @Override
    public String getName() {
        return "Enable Flight";
    }

    @Override
    public String getDescription() {
        return "Enables flight for all players";
    }

    @Override
    public Boolean getSetting() {
        return this.setting;
    }

    @Override
    public void setSetting(Boolean setting) {
        this.setting = setting;
    }

    @Override
    public Material getMenuItemMaterial() {
        return Material.PHANTOM_MEMBRANE;
    }

    @Override
    public int getMenuItemSlot() {
        return 12;
    }

}

File: src/main/java/net/matees/settings/global/FlightEnabled.kt
package net.matees.settings.global

import net.matees.settings.AbstractBooleanSetting
import net.matees.settings.IGlobal
import org.bukkit.Bukkit
import org.bukkit.GameMode
import org.bukkit.Material
import org.bukkit.entity.Player

class FlightEnabled private constructor() : AbstractBooleanSetting(), IGlobal {
    override var setting: Boolean? = false

    override fun onChange() {
        Bukkit.getOnlinePlayers().forEach { player: Player ->
            if (player.gameMode == GameMode.CREATIVE) return@forEach
            player.allowFlight = setting == true
        }
    }

    override val name: String
        get() = "Enable Flight"

    override val description: String
        get() = "Enables flight for all players"

    override val menuItemMaterial: Material
        get() = Material.PHANTOM_MEMBRANE

    override val menuItemSlot: Int
        get() = 12

    companion object {
        val instance: FlightEnabled = FlightEnabled()
    }
}
